### DISCUSSION TITLE
is there a way to restrict image sizes?

### DISCUSSION THREAD

**Discussion by lindamacdonald** [rpldy/react-uploady#208] [General]
Hey there guys :) I was googling how to restrict the image size accepted when uploading an image and I found you. react-uploady looks extremely cool but is there a way to set the allowed max size of an image?

**Comment by yoavniran** [comment]
hey @lindamacdonald 
It depends a little on what you mean by "restrict". 

Using the "cancellable" event with the item start hook - [useItemStartListener](https://github.com/rpldy/react-uploady/tree/master/packages/ui/uploady#useitemstartlistener-event-hook) - you can return false which will make Uploady cancel the upload for that item.
Example code can be found in this [sandbox](https://codesandbox.io/s/react-uploady-cancel-upload-by-image-dimensions-1qvwq).

Let me know if it answers your question.

**Reply by lindamacdonald** [reply]
thank you so much for the sandbox. I love cloudinary but cannot seem to find a custom way of setting image sizes. Let me explain...Have worked with client sites for the better part of 7 years and your average client will not understand that their beautiful Iphone selfies worth several MB will not benefit their websites, so I was wondering if there is a way to set the max size of the file to lets say 500kb per image otherwise give the user a friendly error saying it might not be such a hot idea to upload the image version of Kilimanjaro :)

**Reply by yoavniran** [reply]
Excluding files larger than a certain size is even simpler. You can check this [guide](https://github.com/rpldy/react-uploady/blob/master/guides/FileFilter.md).
If you wish to also notify the users about removing those images from the batch, you could write custom code that updates some state about it and display that info to the user.

You can also expand on the code I shared (in the sandbox) and use the [useItemCancelListener](https://github.com/rpldy/react-uploady/tree/master/packages/ui/uploady#useitemcancellistener-event-hook) hook to update some state for the user notification.

Lastly, at the risk of sounding like a Cloudinary ad :) I can say that there are ways to make the improvements transparent to your customers. For example [incoming transformations](https://cloudinary.com/documentation/transformations_on_upload) or by using [named transformations](https://cloudinary.com/documentation/image_transformations#named_transformations) that will hide some of the complexity, allow to keep the originals, while offering improved performance for the end user. Something to consider.

**Reply by lindamacdonald** [reply]
no go ahead advertise  because as mentioned I love Cloudinary, just fairly new to it so probably dont know half of the tricks ;) I shall go dive deep into the cloudinary docs and if you dont hear from me again I shall still be in the docs hehe :)

thanks a mil for all your kind help
